examples/keras_cv/data_augmentation_with_keras_cv.py

```diff
@@ -0,0 +1,388 @@
+pipeline however you like:
+"""
+
+pipeline = keras_cv.layers.RandomAugmentationPipeline(
+    layers=[keras_cv.layers.GridMask(), keras_cv.layers.Grayscale(output_channels=3)],
+    augmentations_per_image=1,
+)
+
+"""
+This pipeline will either apply `GrayScale` or GridMask:
+"""
+
+
+@tf.function
+def apply_pipeline(inputs):
+    inputs = inputs.copy()
+    inputs["images"] = pipeline(inputs["images"])
+    return inputs
+
+
+train_ds = load_dataset().map(apply_pipeline, num_parallel_calls=tf.data.AUTOTUNE)
+visualize_dataset(train_ds, title="After custom pipeline")
+
+"""
+Looks great!  You can use RandomAugmentationPipeline however you want.
+"""
+
+"""
+# Training a CNN
+
+As a final exercise, let's take some of these layers for a spin.  In this section, we
+will use `CutMix`, `MixUp`, and `RandAugment` to train a state of the art `ResNet50`
+image classifier on the Oxford flowers dataset.
+
+"""
+
+
+def preprocess_for_model(inputs):
+    images, labels = inputs["images"], inputs["labels"]
+    images = tf.cast(images, tf.float32) / 255.0
+    return images, labels
+
+
+train_ds = (
+    load_dataset()
+    .map(apply_rand_augment, num_parallel_calls=tf.data.AUTOTUNE)
+    .map(cut_mix_and_mix_up, num_parallel_calls=tf.data.AUTOTUNE)
+)
+
+visualize_dataset(train_ds, "CutMix, MixUp and RandAugment")
+
+train_ds = train_ds.map(preprocess_for_model, num_parallel_calls=tf.data.AUTOTUNE)
+
+test_ds = load_dataset(split="test")
+test_ds = test_ds.map(preprocess_for_model, num_parallel_calls=tf.data.AUTOTUNE)
+
+train_ds = train_ds.prefetch(5)
+test_ds = test_ds.prefetch(5)
+
+"""
+Next we should create a the model itself.  Notice that we use `label_smoothing=0.1` in
+the loss function.  When using MixUp, label smoothing is _highly_ recommended.
+"""
+
+input_shape = IMAGE_SIZE + (3,)
+
+
+def get_model():
+    model = applications.ResNet50(
+        input_shape=input_shape, classes=num_classes, weights=None
+    )
+    model.compile(
+        loss=losses.CategoricalCrossentropy(label_smoothing=0.1),
+        optimizer=optimizers.SGD(momentum=0.9),
+        metrics=["accuracy"],
+    )
+    return model
+
+
+"""
+Finally we train the model:
+"""
+
+strategy = tf.distribute.MirroredStrategy()
+with strategy.scope():
+    model = get_model()
+    model.fit(
+        train_ds,
+        steps_per_epoch=1000,
+        epochs=10,
+        validation_data=test_ds,
+        validation_steps=50,
+    )
+
+"""
+## Conclusion & Next Steps
+That's all it takes to assemble state of the art image augmentation pipeliens with
+KerasCV!
+
+As an additional exercise for readers, you can:
+
+- Perform a hyper parameter search over the RandAugment parameters to improve the
+classifier accuracy
+- Substitute the Oxford Flowers dataset with your own dataset
+- Experiment with custom `RandomAugmentationPipeline`s.
+
+Currently, between Keras core and KerasCV there are [_28 image augmentation
```

Comment:
All mdown links should be formatted on a single line (cf [guide](https://github.com/keras-team/keras-io/blob/master/contributor_guide.md))